src/main/webapp/app/tasks/task-search/task-search.component.ts

```diff
@@ -0,0 +1,40 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, OnDestroy, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormControl } from '@angular/forms';
+
+import { Subscription } from 'rxjs/internal/Subscription';
+
+@Component({
+  selector: 'tiny-task-search',
+  templateUrl: './task-search.component.html',
+  styleUrls: ['./task-search.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class TaskSearchComponent implements OnInit, OnDestroy {
+  @Output() searched: EventEmitter<string> = new EventEmitter();
+  searchInputFormControl: FormControl;
+  subscriptions: Subscription[] = [];
+
+  constructor(private fb: FormBuilder) { }
+
+  ngOnInit(): void {
+    this.searchInputFormControl = this.fb.control('');
+    this.onSearch();
+  }
+
+  onSearch() {
+    this.subscriptions.push(
+      this.searchInputFormControl.valueChanges.subscribe(searchValue => {
+        searchValue =  searchValue || '';
```

Comment:
in what cases would the `|| '' ` be executed?

Reply:
I thought that it would be work fine in case of user has just entered a space or (enter key) but after trimming it, it's useless now.